ui/src/features/annotator/tools/drawing-box-tool/drawing-box.component.tsx

```diff
@@ -0,0 +1,146 @@
+    Pen = 'pen',
+    Touch = 'touch',
+}
+
+const CURSOR_OFFSET = '7 8';
+interface DrawingBoxInterface {
+    onComplete: (shapes: RectInterface) => void;
+    roi: RegionOfInterest;
+    image: ImageData;
+    zoom: number;
+}
+
+export const DrawingBox = ({ roi, zoom, image, onComplete }: DrawingBoxInterface) => {
+    const [startPoint, setStartPoint] = useState<Point | null>(null);
+    const [boundingBox, setBoundingBox] = useState<RectInterface | null>(null);
+    const [hasCrossHair, setHasCrossHair] = useState<boolean>(true);
+
+    const ref = useRef<SVGRectElement>(null);
+
+    const clampPoint = clampPointBetweenImage(image);
+    const crosshair = useCrosshair(ref, zoom);
+
+    const onPointerMove = (event: PointerEvent<SVGSVGElement>): void => {
+        crosshair.onPointerMove(event);
+
+        if (ref.current === null) {
+            return;
+        }
+
+        const button = {
+            button: event.button,
+            buttons: event.buttons,
+        };
+
+        if (event.pointerType === PointerType.Pen && isEraserButton(button)) {
+            setHasCrossHair(false);
+
+            return;
+        } else {
+            setHasCrossHair(true);
+        }
+
+        if (startPoint === null || !event.currentTarget.hasPointerCapture(event.pointerId)) {
+            return;
+        }
+
+        const endPoint = clampPoint(getRelativePoint(ref.current, { x: event.clientX, y: event.clientY }, zoom));
+
+        setBoundingBox({ shapeType: 'rect', ...clampBox(pointsToRect(startPoint, endPoint), roi) });
+    };
+
+    const onPointerDown = (event: PointerEvent<SVGSVGElement>): void => {
+        if (startPoint !== null || ref.current === null) {
+            return;
+        }
+
+        const button = {
+            button: event.button,
+            buttons: event.buttons,
+        };
+
+        if (event.pointerType === PointerType.Touch || !isLeftButton(button)) {
+            return;
+        }
+
+        const mouse = clampPoint(getRelativePoint(ref.current, { x: event.clientX, y: event.clientY }, zoom));
+
+        event.currentTarget.setPointerCapture(event.pointerId);
+
+        setStartPoint(mouse);
+        setBoundingBox({ shapeType: 'rect', x: mouse.x, y: mouse.y, width: 0, height: 0 });
+    };
+
+    const onPointerUp = (event: PointerEvent<SVGSVGElement>): void => {
+        if (event.pointerType === PointerType.Touch) {
+            return;
+        }
+
+        if (boundingBox === null) {
+            return;
+        }
+
+        // Don't make empty annotations
+        if (boundingBox.width > 1 && boundingBox.height > 1) {
+            onComplete(boundingBox);
+        }
+
+        setCleanState();
+
+        event.currentTarget.releasePointerCapture(event.pointerId);
+    };
+
+    const setCleanState = () => {
+        setStartPoint(null);
+        setBoundingBox(null);
+    };
+
+    useEffect(() => {
+        window.addEventListener('keydown', ({ key }: KeyboardEvent | ReactKeyboardEvent) => {
```

Comment:
This isn’t a priority right now, but we could reuse the `useEventListener` implementation from Geti Classic here in the future.